common/error.cpp

```diff
@@ -48,6 +48,19 @@ std::ostream &operator<<(std::ostream &os, const Error &err) {
 	return os;
 }
 
+Error Error::FollowedBy(const Error &err) const {
+	if (*this == NoError) {
+		return err;
+	}
+	if (err == NoError) {
+		return *this;
+	}
+
+	Error new_err {*this};
+	new_err.message += ": Then followed error: " + err.String();
```

Comment:
I'd use `;` to separate the error messages because `:` often shows up in the messages themselves.